File: src/audioWfc/constraints/hierarchy/NoteLevelNode.java
package audioWfc.constraints.hierarchy;

import audioWfc.HigherValues;
import audioWfc.OctavedNote;
import audioWfc.TileCanvas;
import audioWfc.constraints.CanvasAttributes;
import audioWfc.musicTheory.Note;

import java.util.List;
import java.util.Random;

public class NoteLevelNode {
    private ChordLevelNode parent;
    private HigherValues higherValues;
    private CanvasAttributes<OctavedNote> noteAttributes;
    private TileCanvas<OctavedNote> canvas;

    public NoteLevelNode(ChordLevelNode parent, HigherValues higherValues, CanvasAttributes<OctavedNote> noteAttributes, Random random){
        this.parent = parent;
        this.noteAttributes = noteAttributes;
        this.canvas = new TileCanvas<>(
                noteAttributes.size(),
                noteAttributes.options(),
                noteAttributes.constraints(),
                higherValues,
                random);
    }

    public List<OctavedNote> generate(){
        return canvas.generate();
    }

}

File: vite-project/src/wfc/hierarchy/NoteLevelNode.ts
import { OctavedNote } from "../../music_theory/Note"
import { Random } from "../../util/Random"
import { HigherValues } from "../HigherValues"
import { TileCanvas, TileCanvasProps } from "../TileCanvas"
import { ChordLevelNode } from "./ChordLevelNode"

export class NoteLevelNode {
	private parent : ChordLevelNode
	private canvas : TileCanvas<OctavedNote>

	constructor(parent : ChordLevelNode, canvasProps: TileCanvasProps<OctavedNote>, higherValues: HigherValues, random: Random) {
		this.parent = parent
		this.canvas = new TileCanvas<OctavedNote>(canvasProps, higherValues, random) 
	}

	public generate() : OctavedNote[] {
		return this.canvas.generate()
	}
}